### DISCUSSION TITLE
How to truncate up to n digits after the dot?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [josdejong/mathjs#2328] [Q&A] [answered]
I looked through all issues and the docs, but I couldn't find an answer to my question.

```ts
const config = {
  number: "BigNumber",
  precision: 79,
};
const math = create({ expDependencies }, config);
const mbn = this.math.bignumber!;

function exp(x: string): string {
  const result = mbn(x).exp() as BigNumber;
  console.log(result.toString());
}

exp("-1");
```

The code above logs the following:

```js
0.3678794411714423215955237701614608674458111310317678345078368016974614957448998"
```

Now, I would like to truncate the result to 18 digits *after the dot*. How can I do that? I looked at the [format](https://mathjs.org/docs/reference/functions/format.html) function but it seems like the maximum numbers of digits permitted is 16.

**Comment by PaulRBerg** [comment, marked as answer]
I found the answer myself. math.js is built on top of decimal.js, which has a [toFixed](https://mikemcl.github.io/decimal.js/#toFixed) method. It is similar to the vanilla `toFixed` available in JavaScript, but it can be used only with decimal.js instances.

Thus I would rewrite the example like this:

```ts
function exp(x: string): string {
  const result = mbn(x).exp() as BigNumber;
  console.log(result.toFixed(18));
}
```

And get this in the console:

```
0.367879441171442322
```

**Comment by josdejong** [comment]
There is a powerful function `math.format` that allows you to format numeric values and any type of value into a string

https://mathjs.org/docs/reference/functions/format.html

Your example would look like:

```js
function exp(x: string): string {
  const result = mbn(x).exp() as BigNumber;
  console.log(math.format(result, { notation: 'fixed', precision: 18 }));
}
```

**Reply by PaulRBerg** [reply]
Thanks for the tip! I think that the [docs](https://mathjs.org/docs/reference/functions/format.html) should be updated .. it says that the maximum precision that can be set is 16.

**Reply by josdejong** [reply]
Ah, you're right. Updated via https://github.com/josdejong/mathjs/commit/e6f3123d2041c7f80a75d71227d313a87920806d
